kernel/arm_boards/src/qemu_virt.rs

```diff
@@ -0,0 +1,36 @@
+//! AArch64 Board Config for the `virt` machine of Qemu
```

Comment:
All of these board-specific details should be encapsulated in a struct defined in the `board.rs` module (or in `lib.rs`, e.g.,
```rust
pub struct BoardConfig<const NUM_CPUS: usize> {
    cpu_list: [DefinedCpuId; NUM_CPUS],
    interrupt_controller: InterruptControllerConfig,
    ...
}
```
Then you instantiate it here in this file (pseudo-code-y):

```rust
pub const NUM_CPUS: usize = 4;
pub static ARM_BOARD: BoardConfig = BoardConfig::<NUM_CPUS> {
    cpu_list: [
        DefinedCpuId(0),
        DefinedCpuId(1),
        DefinedCpuId(2),
        DefinedCpuId(3),
    ],
    ...
}
```

You've already done this for the interrupt controller config, so nice job there -- just extend it to everything.